test/e2e/image_scp_test.go

```diff
@@ -0,0 +1,100 @@
+package integration
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/containers/common/pkg/config"
+	. "github.com/containers/podman/v3/test/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("podman image scp", func() {
+	ConfPath := struct {
+		Value string
+		IsSet bool
+	}{}
+	var (
+		tempdir    string
+		podmanTest *PodmanTestIntegration
+	)
+
+	BeforeEach(func() {
+		ConfPath.Value, ConfPath.IsSet = os.LookupEnv("CONTAINERS_CONF")
+		conf, err := ioutil.TempFile("", "containersconf")
+		if err != nil {
+			panic(err)
+		}
+		os.Setenv("CONTAINERS_CONF", conf.Name())
+		tempdir, err = CreateTempDirInTempDir()
+		if err != nil {
+			os.Exit(1)
+		}
+		podmanTest = PodmanTestCreate(tempdir)
+		podmanTest.Setup()
+	})
+
+	AfterEach(func() {
+		podmanTest.Cleanup()
+		os.Remove(os.Getenv("CONTAINERS_CONF"))
+		if ConfPath.IsSet {
+			os.Setenv("CONTAINERS_CONF", ConfPath.Value)
+		} else {
+			os.Unsetenv("CONTAINERS_CONF")
+		}
+		f := CurrentGinkgoTestDescription()
+		processTestResult(f)
+
+	})
+
+	It("podman image scp quiet flag", func() {
+		if IsRemote() {
+			Skip("this test is only for non-remote")
+		}
+		scp := podmanTest.Podman([]string{"image", "scp", "-q", ALPINE})
+		scp.WaitWithDefaultTimeout()
+		Expect(scp).To(Exit(0))
+	})
+
+	It("podman image scp bogus image", func() {
+		if IsRemote() {
+			Skip("this test is only for non-remote")
+		}
+		scp := podmanTest.Podman([]string{"image", "scp", "FOOBAR"})
+		scp.WaitWithDefaultTimeout()
+		Expect(scp).To(ExitWithError())
+	})
+
+	It("podman image scp with proper connection", func() {
+		if IsRemote() {
+			Skip("this test is only for non-remote")
+		}
+		cmd := []string{"system", "connection", "add",
+			"--default",
+			"--identity", "~/.ssh/id_rsa",
+			"QA",
+			"ssh://root@server.fubar.com:2222/run/podman/podman.sock",
+		}
+		session := podmanTest.Podman(cmd)
+		session.WaitWithDefaultTimeout()
+		Expect(session).To(Exit(0))
+
+		cfg, err := config.ReadCustomConfig()
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(cfg.Engine.ActiveService).To(Equal("QA"))
+		Expect(cfg.Engine.ServiceDestinations["QA"]).To(Equal(
+			config.Destination{
+				URI:      "ssh://root@server.fubar.com:2222/run/podman/podman.sock",
+				Identity: "~/.ssh/id_rsa",
+			},
+		))
+
+		scp := podmanTest.Podman([]string{"image", "scp", ALPINE, "QA::"})
+		scp.WaitWithDefaultTimeout()
+		Expect(scp).To(ExitWithError())
```

Comment:
What is this testing? That `QA::` is correctly recognized, but the connection then fails? In that case, the check should somehow verify the “correctly recognized” part.

(That might be too hard to do, I don’t know.)

Reply:
Yeah, testing here is difficult since we cannot do an actual ssh connection. What this does is, it outputs an error at a specific point so I know the code read in and parsed the info correctly but just could not connect, which is the best we can do.

Reply:
Are you saying that the test execution logs an error, and you are manually looking for its presence?

That’s not quite an automated test — but even in that case, please write a comment here saying that this test is executed only for the expected log output, and _what_ is the expected log output.